### ISSUE TITLE
TBB Unordered set - infinite loop in reserve()

### ISSUE THREAD

**Issue by diehard2** [oneapi-src/oneTBB#1138]
I'm seeing an infinite loop in reserve when the size to reserve is small. I did not figure out what the limit is, but 100 works and 26 (my edge case) hangs. I'm on version 2021.9.0 which is the latest offered by vcpkg. This behavior is also present in 2021.8.0

``` 
    tbb::concurrent_unordered_set<std::string_view, XXH3HashStr, std::equal_to<std::string_view>,  tbb::scalable_allocator<std::string_view>> container;
    container.reserve(26);

```

**Comment by kboyarinov** [comment]
Hi @diehard2,

We have fixed similar issue (PR #1057) and changed reserve behavior for the latest oneTBB release. Is it possible to run you example on the latest master?

**Comment by diehard2** [comment]
thanks @kboyarinov I did a cursory search and didn't see anything. I'll have to wait for the next library update on vcpkg, but looks promising!
